src/Summoner/Text.hs

```diff
@@ -0,0 +1,16 @@
+module Summoner.Text
+       ( packageToModule
+       , intercalateMap
+       ) where
+
+import qualified Data.Text as T
+
+-- | Creates module name from the name of the package
+-- Ex: @my-lovely-project@ — @MyLovelyProject@
+packageToModule :: Text -> Text
+packageToModule = T.concat . map T.toTitle . T.splitOn "-"
```

Comment:
You can move `headToUpper` function to this module as well and use it here instead of `toTitle`.